bot.py

```diff
@@ -29,492 +20,75 @@
-            rmsg(response, ["Hello, " + temp_user["user"]["profile"]["first_name"] + "! :tada:"])  # noqa: 501
-            return True
-        elif response["text"].lower() == "pantherbot ping":
-            rmsg(response, ["PONG"])
-            return True
-        elif response["text"].lower() == ":rip: pantherbot" or response["text"].lower() == "rip pantherbot":  # noqa: 501
-            rmsg(response, [":rip:"])
-            return True
-        elif re.match(".*panther +hackers.*", str(response["text"].lower())):
-            rmsg(response, ["NO THIS IS PANTHERHACKERS"])
-            return True
-        elif "subtype" in response:
-            if response["subtype"] == "channel_leave" or response["subtype"] == "group_leave":  # noqa: 501
-                rmsg(response, ["Press F to pay respects"])
-                return True
-        return False
-    except:
-        print "Error with checking in other_message: likely the message contained unicode characters"
-
-
-def riyans_denial(response):
-    if "U0LJJ7413" in response["user"]:
-        if response["text"][:1] in ["!", "$"] or response["text"].lower() in ["hey pantherbot", "pantherbot ping"]:  # noqa: 501
-            rmsg(response, ["No."])
-            return True
-    return False
-
-
-def new_user_message(response):
-    print "PantherBot:LOG:Member joined team"
-    sc.api_call(
-        "chat.postMessage",
-        channel=response["user"]["id"],
-        text=GREETING,
-        username=BOT_NAME,
-        icon_url=BOT_ICON_URL
-    )
-
-
-def react_announcement(response):
-    if GENERAL != "" and response["channel"] == GENERAL:
-        temp_list = list(EMOJI_LIST)
-        rreaction(response, "pantherbot")
-        for x in range(0, 3):
-            num = random.randrange(0, len(temp_list))
-            rreaction(response, temp_list.pop(num))
-
-
-# Less used WebSocket functions
-def on_error(ws, error):
-    print "PantherBot:LOG:ERROR"
-    print error
-
-
-def on_close(ws):
-    print "PantherBot:LOG:Connection lost or closed..."
-
-
-# send a response message (sends to same channel as command was issued)
-def rmsg(response, l):
-    for text in l:
-        sc.api_call(
+        pri_channels = self.SC.api_call(
+            "groups.list",
+            exclude_archived=1
+        )
+        li = []
+        for channel in pub_channels["channels"]:
+            for num in range(0, len(channel_names)):
+                if channel["name"].lower() == channel_names[num].lower():
+                    li.append(channel["id"])
+        # Same as above
+        for channel in pri_channels["groups"]:
+            for num in range(0, len(channel_names)):
+                if channel["name"].lower() == channel_names[num].lower():
+                    li.append(channel["id"])
+        return li
+
+    # Send Message
+    # Sends a message to the specified channel (looks up by channel name, unless is_id is True)
+    def smsg(self, channel, text, is_id=False):
```

Comment:
i'm assuming this is `send_msg`? Should the name be changed to that to be more clear?